src/components/shared/Snackbar/index.tsx

```diff
@@ -38,52 +41,95 @@ const ActionButton = styled.View`
 `;
 
 export interface SnackbarProps {
+  testID?: string;
+  ref?: any;
+}
+
+interface Content {
   text: string;
   actionText?: string;
-  testID?: string;
-  show: boolean;
-  setShow: (show: boolean) => void;
   timer?: Timer;
   actionStyle?: TextStyle;
   containerStyle?: ViewStyle;
   messageStyle?: TextStyle;
   onPressAction?: () => void;
 }
 
+interface ShowingState {
+  isVisible?: boolean;
+  isShowing?: boolean;
+  timeout?: any;
+}
+
+export interface SnackbarRef {
+  show(content: Content): void;
+}
+
 export enum Timer {
   SHORT = 1500,
   LONG = 3000,
 }
 
-const Snackbar: React.FC<SnackbarProps> = (props) => {
-  const { testID, show, setShow, timer = Timer.SHORT, actionText } = props;
+const Snackbar: React.FC<SnackbarProps> = React.forwardRef<SnackbarRef, SnackbarProps>((props, ref) => {
+  const { testID } = props;
+  const [showingState, setShowingState] = React.useState<ShowingState>({ isVisible: false, isShowing: false });
+  const [content, setContent] = React.useState<Content>({ text: '', timer: Timer.SHORT });
+  const { text, actionText, messageStyle, actionStyle, containerStyle, timer = Timer.SHORT, onPressAction } = content;
+  const { isShowing, isVisible, timeout } = showingState;
+  const [fadeAnim] = React.useState(new Animated.Value(0));
+  const show = (content): void => {
+    setContent(content);
+    clearTimeout(timeout);
+    setShowingState({ isShowing: true });
+  };
+  const close = (): void => {
+    Animated.timing(
+      fadeAnim,
+      {
+        toValue: 0,
+        duration: 200,
+      },
+    ).start(() => setShowingState({ isVisible: false }));
+  };
   React.useEffect(() => {
-    let timeout;
-    if (show === true) {
-      timeout = setTimeout(() => {
-        setShow(false);
-      }, timer);
+    if (isShowing) {
+      if (isVisible) {
+        close();
+      } else {
```

Comment:
I don't see in which situation that case will be met, could you elaborate a bit for me please?

Reply:
seems like there is no `isShowing === true && isVisible ===true` case in the whole process

Reply:
You are right. It was my mistake. I'll fix it.